code/src/reducers/tasks.js

```diff
@@ -0,0 +1,29 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const tasks = createSlice({
+    name: 'tasks',
+    initialState: {
+        items: [],
+    },
+    reducers: {
+        toggleItem: (store, action) => {      
+            store.items.forEach(item => {
+                if (item.id === action.payload) {
+                    item.complete = !item.complete
+                }
+            })
+        },
+        //Mutable reducer function to use splice and push but works since immer included in dependencies
```

Comment:
I like that you included this comment - I was forgetting myself that immer is a dependency. It's great for future reference. After I watched this week's videos about how you can also write reducers, I'm so happy we have immer and reduxjs toolkit to help us 😅